projects/plugins/jetpack/extensions/shared/components/product-management-controls/toolbar-control.js

```diff
@@ -68,7 +68,7 @@ function NewProduct( { onClose } ) {
 			<MenuItem>
 				{ siteSlug && (
 					<ExternalLink
-						href={ `https://wordpress.com/earn/payments-plans/${ siteSlug }#add-new-payment-plan` }
+						href={ `https://wordpress.com/earn/payments/${ siteSlug }#add-new-payment-plan` }
```

Comment:
Tested that the hash works with the redirect [at the other PR](https://github.com/Automattic/wp-calypso/pull/84574/files#diff-2a6dfbe3a18e5985c24b5043060567f36d6817938470e5bd68b665b46ccff756R13-R15) 👍